src/mon/ConnectionTracker.h

```diff
@@ -185,6 +185,18 @@ class ConnectionTracker {
     rank = new_rank;
     my_reports.rank = rank;
   }
+  /**
+  * Check if our ranks are clean.
+  * In the future we also want to check the reports
+  * current and history of each peer_report.
+  */
+  bool is_clean(int mon_rank) {
+    if (rank != mon_rank ||
+      my_reports.rank != mon_rank) {
```

Comment:
Is this the condition we're always failing? I thought it could also just involve having some extra peer_reports in the map.

Reply:
Yeah; I'm worried about the case where some monitors have an old rank 6 score sitting around and others don't.

Reply:
Addressed in new push